src/expr/skolem_manager.h

```diff
@@ -132,74 +132,187 @@ enum class SkolemFunId
    * i = 0, ..., n.
    */
   RE_UNFOLD_POS_COMPONENT,
-  BAGS_CARD_CARDINALITY,
+  /**
+   * An uninterpreted function for bag.fold operator:
+   * To compute (bag.card A), we need a function that
+   * counts multiplicities of distinct elements. We call this function
+   * combine of type Int -> Int where:
+   * combine(0) = 0.
+   * combine(i) = m(elements(i), A) + combine(i-1) for 1 <= i <= n.
+   * elements: a skolem function for (bag.fold f t A)
+  *            see BAGS_CARD_ELEMENTS.
+   * n: is the number of distinct elements in A.
+   *
+   * - Number of skolem arguments: ``1``
+   *   - ``1:`` an integer index in [0,n].
```

Comment:
I don't think this is accurate.

The skolem arguments denote what this skolem is indexed by, which I believe in this case is a bag (the bag to apply cardinality to).

In other words, we are describing the expected `cacheVals` when constructing this function.

Reply:
Ok, I understood it as the arguments to the functions, not the indexing arguments. 
I am missing the context, what is the goal of this documentation?

Reply:
We are exporting the skolems in proofs and lemmas which users can inspect. Thus, they need full documentation on how they are created.  We potentially will allow users to create these skolems, thus we need to indicate how many arguments they expect.